08-CMSIS/Makefile

```diff
@@ -0,0 +1,128 @@
+
+
+
+# Target:STM32F429 device
+STM32F429                       := STM32F429
+STM32F429_RELEASE_DIR           := $(RELEASE_DIR)/$(STM32F429)
+STM32F429_PLATFORM_DIR          := $(PLATFORM_DIR)/$(STM32F429)
+STM32F429_CMSIS_DIR             := $(CMSIS_DIR)/$(STM32F429)
+
+STM32F429_DEPENDENT_HEADER_DIR  := $(STM32F429_PLATFORM_DIR)/$(INC_DIR)/
+STM32F429_DEPENDENT_SOURCE      := $(wildcard $(STM32F429_PLATFORM_DIR)/$(SRC_DIR)/*.c)
+STM32F429_INC_DIR               := $(CORE_HEADER_DIR) $(STM32F429_DEPENDENT_HEADER_DIR) $(CMSIS_DIR) $(STM32F429_CMSIS_DIR)
+
+STM32F429_TARGET_name           := $(STM32F429)_os
+STM32F429_TARGET                := $(STM32F429_TARGET_name).bin
+STM32F429_TARGET_ld             := $(STM32F429_PLATFORM_DIR)/os.ld
+STM32F429_TARGET_bin            := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).bin
+STM32F429_TARGET_elf            := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).elf
+STM32F429_TARGET_dump           := $(STM32F429_RELEASE_DIR)/$(STM32F429_TARGET_name).list
+###################
+
+
+all: $(STM32P103_TARGET) $(STM32F429_TARGET)
+
+
+$(STM32P103_TARGET): $(CORE_SOURCE) $(STM32P103_DEPENDENT_SOURCE)
+	mkdir -p $(STM32P103_RELEASE_DIR)
+	$(CC) $(CFLAGS) -mcpu=$(CORTEX_M3) -Wl,-T$(STM32P103_TARGET_ld) -nostartfiles \
+		  $(addprefix -I,$(STM32P103_INC_DIR)) \
+		  $^ -o $(STM32P103_TARGET_elf)
+	$(CROSS_COMPILE)objcopy -Obinary $(STM32P103_TARGET_elf) $(STM32P103_TARGET_bin)
+	$(CROSS_COMPILE)objdump -S $(STM32P103_TARGET_elf) > $(STM32P103_TARGET_dump)
+
+
+
+$(STM32F429_TARGET): $(CORE_SOURCE) $(STM32F429_DEPENDENT_SOURCE)
+	mkdir -p $(STM32F429_RELEASE_DIR)
+	$(CC) $(CFLAGS) -mcpu=$(CORTEX_M4) -Wl,-T$(STM32F429_TARGET_ld) -nostartfiles \
+		  $(addprefix -I,$(STM32F429_INC_DIR)) \
+		  $^ -o $(STM32F429_TARGET_elf)
+	$(CROSS_COMPILE)objcopy -Obinary $(STM32F429_TARGET_elf) $(STM32F429_TARGET_bin)
+	$(CROSS_COMPILE)objdump -S $(STM32F429_TARGET_elf) > $(STM32F429_TARGET_dump)
+
+
+
+flash: $(STM32F429_TARGET)
+	st-flash --reset write $(STM32F429_RELEASE_DIR)/$^ 0x8000000
+
+
+erase:
+	st-flash erase
+
+
+gdb_ST-UTIL: $(STM32F429_TARGET_elf)
+	@echo ""
+	@echo "Before execute this command,you should:"
+	@echo "Open another terminal,and execute the command:\"st-util\""
+	@echo ""
+	arm-none-eabi-gdb $^ -ex "target remote:4242"
+
+
+qemu: $(STM32P103_TARGET)
+	@qemu-system-arm -M ? | grep stm32-p103 >/dev/null || exit
+	@echo "Press Ctrl-A and then X to exit QEMU"
+	@echo
+	qemu-system-arm -M stm32-p103 -nographic -kernel $(STM32P103_TARGET_bin)
+
+
+qemu_GDBstub: $(STM32P103_TARGET)
+	@echo ""
+	@echo "Open another terminal,and type \"make qemu_GDBconnect\""
+	@echo ""
+	qemu-system-arm -M stm32-p103 -nographic -kernel $(STM32P103_TARGET_bin) -s -S
+
+
+qemu_GDBconnect: $(STM32P103_TARGET_elf)
+	arm-none-eabi-gdb $^ -ex "target remote:1234"
+
+
+cscope:
+	cscope -Rbqf ./cscope.out
+
+astyle:
+	@echo "More details please see:https://github.com/jserv/mini-arm-os/blob/master/coding-style.txt"
```

Comment:
Don't put URL in output messages since it is not console-friendly. Instead, you can point out the path, `../coding-style.txt`.